From c87440540f94c1b5c8ea10761ca657830b268e0e Mon Sep 17 00:00:00 2001
From: Eugeniy Paltsev <Eugeniy.Paltsev@synopsys.com>
Date: Fri, 12 Jan 2018 13:44:24 +0300
Subject: [PATCH 61/62] ARC: HSDK: Setup AXI apertures unconditionally

Setup AXI apertures unconditionally as we want to have DDR
in 0x00000000 region when we are kicking slave cpus.

Signed-off-by: Eugeniy Paltsev <Eugeniy.Paltsev@synopsys.com>
---
 board/synopsys/hsdk/hsdk.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/board/synopsys/hsdk/hsdk.c b/board/synopsys/hsdk/hsdk.c
index b2f9ef3cae..f97c010eb8 100644
--- a/board/synopsys/hsdk/hsdk.c
+++ b/board/synopsys/hsdk/hsdk.c
@@ -16,6 +16,9 @@ DECLARE_GLOBAL_DATA_PTR;
 #define	CREG_PAE_UPDATE	(CREG_BASE + 0x194)
 #define	CREG_CPU_START	(CREG_BASE + 0x400)
 
+/* TODO: move AXI apertures setup code here from hsdk-cmd.c */
+void init_memory_bridge(void);
+
 int board_early_init_f(void)
 {
 	/* In current chip PAE support for DMA is broken, disabling it. */
@@ -24,6 +27,10 @@ int board_early_init_f(void)
 	/* Really apply settings made above */
 	writel(1, (void __iomem *) CREG_PAE_UPDATE);
 
+	/* Setup AXI apertures unconditionally as we want to have DDR
+	 * in 0x00000000 region when we are kicking slave cpus */
+	init_memory_bridge();
+
 	return 0;
 }
 
-- 
2.11.0

